### ISSUE TITLE
Enable support for ShareNotify v2

### ISSUE THREAD

**Issue by awead** [psu-libraries/scholarsphere-3#622]
ShareNotify's API is now at version 2. Scholarsphere is trying to POST version 1 requests to Share which aren't supported anymore.

The ShareNotify gem needs to be updated to support v2. See https://github.com/projecthydra-labs/share_notify/issues/11

Once that is complete, we can update the gem in Scholarsphere and re-export all our context back to Share.

**Comment by awead** [comment]
There is conflicting information about which API endpoint to use. It seems that this is the correct one:

https://share.osf.io/api/v2/

However, the test endpoint does not match this because it's missing the /normalizeddata endpoint:

https://api.test.osf.io/v2/

This possibly could be a new "production" endpoint because its response matches the test one:

https://api.test.osf.io/v2/

**Comment by carolyncole** [comment]
Waiting for a response from Share Notify to verify our approach here: https://github.com/psu-stewardship/scholarsphere/tree/i%23622

**Comment by mtribone** [comment]
won't fix.
